packages/http-client/src/contracts/__tests__/type-checks.test-d.ts

```
@@ -0,0 +1,399 @@
+    });
+});
+
+describe("No any types leak through", () => {
+    test("client methods don't return any", () => {
+        const client = {} as HttpClient;
+
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        expectTypeOf(client.get<unknown>).parameters.not.toEqualTypeOf<[any]>();
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        expectTypeOf(client.post<unknown>).parameters.not.toEqualTypeOf<
+            [any]
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)